esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2008,7 +2118,7 @@ bool CFileSprayEx::onSprayFixed(IEspContext &context, IEspSprayFixed &req, IEspS
         wu->setCommand(DFUcmd_import);
 
         IDFUfileSpec *source = wu->queryUpdateSource();
-        checkDZScopeAccessAndSetSpraySourceDFUFileSpec(context, sourcePlaneReq, sourceIPReq, srcfile, srcxml, source);
+        setSpraySourceDFUFileSpec(sourceHostReq, sourcePathReq, srcxml, source);
```

Comment:
This is probably going to need revising to ensure it includes plane name, following this change: https://github.com/hpcc-systems/HPCC-Platform/pull/17353/files#diff-8b47ea38bf0ea671d5db7b9ae0166a6c19a1ad525011ffac7556726d8245844a

And for that reason and others this PR should be rebased onto master.

Reply:
>Not sure why we need to 'ensure it includes plane name'? Do we want to call the source->setGroupName(sourcePlaneReq); inside the setSpraySourceDFUFileSpec()?

See https://track.hpccsystems.com/browse/HPCC-29553
It ensures the group names are created (in bare-metal) for planes, meaning the logical files (IFileDescriptor's) will have properly formed group not IP's within them.
The IFileDescriptor's created by DFU need to know the group(plane) name, and hence this (setGroupName) is the means by which they do.

Reply:
>If this PR is rebased onto master, should we create another small 9.0.x PR to fix the crash when isDropZoneRestrictionEnabled = false and getAndValidateDropZone() returns nullptr and to not call getDZFileScopePermissions() if no plane name is found?

Yes, that makes sense.